ui/selectors/selectors.js

```diff
@@ -82,8 +82,9 @@ export function getCurrentKeyring(state) {
 }
 
 export function isEIP1559Account(state) {
-  // Neither hardware wallet supports 1559 at this time
-  return !isHardwareWallet(state);
+  // Only Ledger supports 1559 at this times
+  const currentKeyring = getCurrentKeyring(state);
+  return currentKeyring && currentKeyring.type !== 'Trezor Hardware';
```

Comment:
Nit: this literal `'Trezor Hardware'` is used in a few places throughout the codebase (defined as `KEYRING_TYPES.TREZOR` in `MetamaskController`) this would be a good candidate for a shared constant I think